firewood/src/db.rs

```diff
@@ -305,13 +306,64 @@ impl<S: ShaleStore<Node> + Send + Sync> api::DbView for DbRev<S> {
             .map_err(|e| api::Error::IO(std::io::Error::new(ErrorKind::Other, e)))
     }
 
-    async fn range_proof<K: api::KeyType, V, N>(
+    async fn range_proof<K: api::KeyType, V>(
```

Comment:
This should really be implemented in `merkle.rs` and tested there as well. You should be able to generate a `RangeProof` on `Merkle` and the `Db::range_proof` should just delegate.

Reply:
+1